desktop-core/src/main/java/com/stnetix/ariaddna/desktopcore/fileWatchigService/Consumer.java

```diff
@@ -0,0 +1,155 @@
+     *
+     * @param events list ov native events from native WatchService
+     * @param dir    directory in which the events occurred
+     */
+    void adaptEvent(List<WatchEvent<?>> events, Path dir) {
+        for (int i = 0; i < events.size(); i++) {
+            //get native event
+            @SuppressWarnings("rawtypes")
+            WatchEvent<Path> event = (WatchEvent<Path>) events.get(i);
+
+            //get type of event
+            WatchEvent.Kind kind = event.kind();
+
+            // Context for directory entry event is the file name of entry
+            @SuppressWarnings("unchecked")
+            Path name = event.context();
+            Path child = dir.resolve(name);
+
+            //Check for MODIFY events
+            if (kind.equals(ENTRY_MODIFY)) {
+                service.runEventListeners(new FileSystemWatchEvent(FileSystemWatchEvent.Type.MODIFY, child));
+                continue;
+            }
+
+            //determination RENAME events
+            if (events.size() > 1) {
+                if (i < events.size() - 1) {
+                    @SuppressWarnings("rawtypes")
+                    WatchEvent<Path> nextEvent = (WatchEvent<Path>) events.get(i + 1);
+                    WatchEvent.Kind nextKind = nextEvent.kind();
+                    Path nextName = nextEvent.context();
+                    Path nextChild = dir.resolve(nextName);
+                    if (kind.name().equals("ENTRY_DELETE") && nextKind.name().equals("ENTRY_CREATE") &&
+                            child.getParent().equals(nextChild.getParent())) {
+                        //System.out.println("RENAME from "+child+" to "+nextChild);
+                        service.runEventListeners(new FileSystemWatchEvent(FileSystemWatchEvent.Type.RENAME, child, nextChild));
+                        i++;
+                        continue;
+                    }
+                }
+                //if event is not RENAME add it to the queue
+                createQueue.add(new InnerWatchEvent(child, event));
+            } else createQueue.add(new InnerWatchEvent(child, event));
+
+            // print out event
+            //System.out.format("%s: %s\n", event.kind().name(), child);
+
+            // if directory is created, and watching recursively, then register it and its sub-directories
+            if (kind == ENTRY_CREATE) {
+                try {
+                    if (Files.isDirectory(child)) {
+                        service.walkAndRegisterDirectories(child);
+                    }
+                } catch (IOException x) {
+                    // do something useful
+                }
+            }
+        }
+    }
+
+    /**
+     * Thread that scans all raw events from queue and determine other events
+     */
+    @Override
+    public void run() {
+
+        while (!isStoped) {
+            try {
+                Thread.sleep(20);
+            } catch (InterruptedException e) {
+                e.printStackTrace();
```

Comment:
This exception should be write to log on debug level "service XXX was stopped" and only on trace level print stacktrace